run/types/DeviceWrapper.ts

```diff
@@ -2480,28 +2507,73 @@ export class DeviceWrapper {
       selector: 'Modal description',
     } as StrategyExtractionObj;
 
-    // New → legacy fallback
+    // Locators
     const elHeading = await this.findWithFallback(newHeading, legacyHeading);
     const elDescription = await this.findWithFallback(newDescription, legacyDescription);
 
-    // Modal Heading
-    const actualHeading = removeNewLines(await this.getTextFromElement(elHeading));
-    if (expectedHeading === actualHeading) {
-      this.log('Modal heading is correct');
-    } else {
-      throw new Error(
-        `Modal heading is incorrect.\nExpected: ${expectedHeading}\nActual: ${actualHeading}`
-      );
+    // Actual text
+    const actualHeading = await this.getTextFromElement(elHeading);
+    const actualDescription = await this.getTextFromElement(elDescription);
+
+    this.assertTextMatches(actualHeading, expectedHeading, 'Modal heading');
+    this.assertTextMatches(actualDescription, expectedDescription, 'Modal description');
+  }
+
+  /**
+   * Checks CTA component text against expected values.
+   * CTAs contain: heading, body, 0-3 features, 1-2 buttons.
+   * @param heading - Expected CTA heading text
+   * @param body - Expected CTA body text
+   * @param buttons - Expected button text(s). First is positive, second (if present) is negative
+   * @param features - Optional array of expected feature text (0-3 items)
+   * @throws Error if any text element doesn't match expected value
+   */
+  public async checkCTAStrings(
+    heading: string,
+    body: string,
+    buttons: string[],
+    features?: string[]
+  ): Promise<void> {
+    // Validate input
+    if (features && features.length > 3) {
+      throw new Error('CTAs support maximum 3 features');
+    }
+    if (buttons.length < 1 || buttons.length > 2) {
+      throw new Error('CTAs must have 1-2 buttons');
+    }
+
+    // Find and check heading
+    const elHeading = await this.waitForTextElementToBePresent(new CTAHeading(this));
+    const actualHeading = await this.getTextFromElement(elHeading);
+    this.log(actualHeading);
```

Comment:
if we still want this, maybe add a prefix so we know what gets printed

```suggestion
    this.log(`actualHeading: `,actualHeading);
```

But the next line might already be printing it?